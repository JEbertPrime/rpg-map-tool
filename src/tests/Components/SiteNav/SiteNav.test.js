import { render, screen, mount } from "@testing-library/react";
import Nav, {LinkWrap} from "../../../../Components/SiteNav";
import renderer from 'react-test-renderer';
describe('LinkWrap', ()=>{
  it("passes on refs",()=>{
    const testProps = {refAs: true}
    render(<LinkWrap {...testProps}><a>test</a></LinkWrap>)
    expect(
      screen.getByText('test')
    ).toBeInTheDocument()

  });
  it('filters invalid children', ()=>{
    const testProps = {refAs: true}
    render(<LinkWrap {...testProps}>dog</LinkWrap>)
    expect(
      screen.getByText('invalid')
    ).toBeInTheDocument()
  })
})
const testSession = [{user:{image:'/maps/example_1.jpg', name: 'Johnny'}},{user:{ name: 'Johnny Testworth'}}]
describe("Nav", () => {
  it("renders without crashing", () => {
    render(<Nav />);
    expect(
      screen.getByRole("navigation")
    ).toBeInTheDocument();
    expect(
      screen.getByRole('link', {name: 'campaignMaps'})
    ).toBeInTheDocument();
    expect(
      screen.getByRole('link', {name: 'map'})
    ).toBeInTheDocument();
    
  });
  it('renders user session image correctly', ()=>{
    render(<Nav session={testSession[0]}/>)
    expect(
      screen.getByAltText('user image')
    ).toBeInTheDocument();
  })
  it('renders user session name correctly', ()=>{
    render(<Nav session={testSession[1]}/>)
    expect(
      screen.getByText('Johnny Testworth')
    ).toBeInTheDocument();
  })
  it("renders correctly", ()=>{
    const tree = renderer.create(<Nav />).toJSON();
    expect(tree).toMatchSnapshot();

  })
});