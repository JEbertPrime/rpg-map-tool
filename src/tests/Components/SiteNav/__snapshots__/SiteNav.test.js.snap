// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders correctly 1`] = `
<div>
  <nav
    className="navbar navbar-expand-md navbar-light bg-white"
    id="main_nav"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      campaignMaps
    </a>
    <button
      aria-label="Toggle navigation"
      className="navbar-toggler"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      style={Object {}}
    >
      <ul
        className="me-auto navbar-nav"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/api/auth/signout/"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            signout
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/map/"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            map
          </a>
        </li>
      </ul>
    </div>
    <a
      className="nav-link"
      href="/api/auth/signin/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      login
    </a>
  </nav>
</div>
`;
